pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -175,18 +175,17 @@ func (m *manager) AddContainer(p *v1.Pod, c *v1.Container, containerID string) e
 		return err
 	}
 	cpus := m.state.GetCPUSetOrDefault(containerID)
-	m.Unlock()
+	defer m.Unlock()
```

Comment:
The previous code avoided holding the lock while doing the I/O involved in updateContainerCPUSet (this involves a gRPC call to the CRI.) Could we release and re-acquire the lock if necessary before the RemoveContainer call?

Reply:
will update it.